lib/friends/introvert.rb

```diff
@@ -677,6 +688,7 @@ def read_file
         # Parse the line and update the parsing state.
         state = parse_line!(line, line_num: line_num, state: state)
       end
+      set_implicit_locations
```

Comment:
Ah, this is a minor thing but I've been using bangs (`!`) as a signal that objects are being mutated. Since this is setting a field on Locations, would you mind making this `set_implicit_locations!`?

Reply:
I read about bangs and `?` just the other day :) Makes sense and done!

Reply:
Weird - I made the change and it passes on my local, but not in Travis. I looked at the failing test and I can't quite figure out why it would fail after adding a bang to `set_implicit_locations` 🤔 Do you think you could have a look? I've reverted the commit for this for now while we figure out what is going on